suit_generator/cmd_encrypt.py

```diff
@@ -0,0 +1,222 @@
+        required=True,
+        type=auto_int,
+        help="Key ID used to identify the key on the device.",
+    )
+    cmd_generate_info_parser.add_argument(
+        "--kw-alg",
+        default=SuitKWAlgorithms.DIRECT.value,
+        type=SuitKWAlgorithms,
+        choices=list(SuitKWAlgorithms),
+        help="Key wrap algorithm used to wrap the CEK.",
+    )
+    cmd_generate_info_parser.add_argument(
+        "--output-dir", required=True, type=Path, help="Directory to store the output files"
+    )
+
+    cmd_generate_info_parser.add_argument(
+        "--encrypt-script",
+        required=True,
+        help="Encrypt script used to generate the encryption artifacts. "
+        + "It must contain a function suit_encryptor_factory() returning an object implementing SuitEncryptorBase.",
+    )
+
+
+def encrypt_and_generate(**kwargs):
+    """Encrypt the payload and generate the files."""
+    encryptor = _import_encryptor(kwargs["encrypt_script"])
+    with open(kwargs["firmware"], "rb") as file:
+        plaintext = file.read()
+    encrypted_content, tag, encryption_info, digest, plaintext_len = encryptor.encrypt_and_generate(
+        plaintext,
+        kwargs["key_name"],
+        kwargs["key_id"],
+        kwargs["context"],
+        kwargs["hash_alg"],
+        kwargs["kw_alg"],
+        kwargs["kms_script"],
+    )
+    with open(os.path.join(kwargs["output_dir"], "plain_text_digest.bin"), "wb") as file:
+        file.write(digest)
+    with open(os.path.join(kwargs["output_dir"], "plain_text_size.txt"), "w") as file:
+        file.write(str(plaintext_len))
+    with open(os.path.join(kwargs["output_dir"], "suit_encryption_info.bin"), "wb") as file:
+        file.write(encryption_info)
+    with open(os.path.join(kwargs["output_dir"], "encrypted_content.bin"), "wb") as file:
+        file.write(tag + encrypted_content)
+
+
+def generate_info(**kwargs):
+    """Generate files based on encrypted firmware and the encrypted content/asset encryption key."""
+    encryptor = _import_encryptor(kwargs["encrypt_script"])
+    with open(kwargs["encrypted_firmware"], "rb") as file:
+        encrypted_firmware = file.read()
+    with open(kwargs["encrypted_key"], "rb") as file:
+        encrypted_key = file.read()
+    encrypted_content, tag, encryption_info = encryptor.generate(
+        encrypted_firmware,
+        encrypted_key,
+        kwargs["key_id"],
+        kwargs["kw_alg"],
+    )
+    with open(os.path.join(kwargs["output_dir"], "suit_encryption_info.bin"), "wb") as file:
+        file.write(encryption_info)
+    with open(os.path.join(kwargs["output_dir"], "encrypted_content.bin"), "wb") as file:
+        file.write(tag + encrypted_content)
+
+
+def main(**kwargs) -> None:
+    """Sign a SUIT envelope."""
+    if kwargs["encrypt_subcommand"] == ENCRYPT_AND_GENERATE_FIRMWARE_CMD:
+        encrypt_and_generate(**kwargs)
+    elif kwargs["encrypt_subcommand"] == GENERATE_INFO_FIRMWARE_CMD:
+        generate_info
```

Comment:
Was it intended as a function call? Missing parentheses?
If not, then a comment explaining intention would be useful.